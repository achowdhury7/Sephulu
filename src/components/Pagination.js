import React, { Component } from 'react'
import { Pager } from 'react-bootstrap'

const Pagination = (props) => {
  return (
    <Pager onSelect={props.handleSelect}>
      <Pager.Item href="#" eventKey="prev" disabled={!props.hasPrev}>Previous</Pager.Item>
      <Pager.Item href="#" eventKey="next" disabled={!props.hasNext}>Next</Pager.Item>
    </Pager>
  )
}

export default Pagination
