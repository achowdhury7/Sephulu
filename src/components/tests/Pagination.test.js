import React from 'react'
import { shallow, mount, render } from 'enzyme'

import Pagination from '../Pagination'

test('render a pagination bar', () => {
  const wrapper = shallow(
    <Pagination />
  )
  expect(wrapper).toMatchSnapshot()
})

test('disable `previous` button if `hasPrev` is false', () => {
  const wrapper = mount(
    <Pagination hasPrev={false} />
  )

  expect(wrapper).toMatchSnapshot()

  expect(wrapper.findWhere(n => n.prop('eventKey') === 'prev').prop('disabled')).toEqual(true)
})

test('disable `next` button if `hasNext` is false', () => {
  const wrapper = mount(
    <Pagination hasNext={false} />
  )

  expect(wrapper).toMatchSnapshot()

  expect(wrapper.findWhere(n => n.prop('eventKey') === 'next').prop('disabled')).toEqual(true)
})