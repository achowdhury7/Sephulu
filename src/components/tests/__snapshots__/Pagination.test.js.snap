// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disable \`next\` button if \`hasNext\` is false 1`] = `
<Pagination
  hasNext={false}
>
  <Pager
    bsClass="pager"
  >
    <ul
      className="pager"
    >
      <PagerItem
        disabled={true}
        eventKey="prev"
        href="#"
        key=".0"
        next={false}
        onSelect={null}
        previous={false}
      >
        <li
          className="disabled"
        >
          <SafeAnchor
            componentClass="a"
            disabled={true}
            href="#"
            onClick={[Function]}
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "pointerEvents": "none",
                }
              }
              tabIndex={-1}
            >
              Previous
            </a>
          </SafeAnchor>
        </li>
      </PagerItem>
      <PagerItem
        disabled={true}
        eventKey="next"
        href="#"
        key=".1"
        next={false}
        onSelect={null}
        previous={false}
      >
        <li
          className="disabled"
        >
          <SafeAnchor
            componentClass="a"
            disabled={true}
            href="#"
            onClick={[Function]}
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "pointerEvents": "none",
                }
              }
              tabIndex={-1}
            >
              Next
            </a>
          </SafeAnchor>
        </li>
      </PagerItem>
    </ul>
  </Pager>
</Pagination>
`;

exports[`disable \`previous\` button if \`hasPrev\` is false 1`] = `
<Pagination
  hasPrev={false}
>
  <Pager
    bsClass="pager"
  >
    <ul
      className="pager"
    >
      <PagerItem
        disabled={true}
        eventKey="prev"
        href="#"
        key=".0"
        next={false}
        onSelect={null}
        previous={false}
      >
        <li
          className="disabled"
        >
          <SafeAnchor
            componentClass="a"
            disabled={true}
            href="#"
            onClick={[Function]}
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "pointerEvents": "none",
                }
              }
              tabIndex={-1}
            >
              Previous
            </a>
          </SafeAnchor>
        </li>
      </PagerItem>
      <PagerItem
        disabled={true}
        eventKey="next"
        href="#"
        key=".1"
        next={false}
        onSelect={null}
        previous={false}
      >
        <li
          className="disabled"
        >
          <SafeAnchor
            componentClass="a"
            disabled={true}
            href="#"
            onClick={[Function]}
          >
            <a
              href="#"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              style={
                Object {
                  "pointerEvents": "none",
                }
              }
              tabIndex={-1}
            >
              Next
            </a>
          </SafeAnchor>
        </li>
      </PagerItem>
    </ul>
  </Pager>
</Pagination>
`;

exports[`render a pagination bar 1`] = `
<Pager
  bsClass="pager"
>
  <PagerItem
    disabled={true}
    eventKey="prev"
    href="#"
    next={false}
    previous={false}
  >
    Previous
  </PagerItem>
  <PagerItem
    disabled={true}
    eventKey="next"
    href="#"
    next={false}
    previous={false}
  >
    Next
  </PagerItem>
</Pager>
`;
